src/render/filters.rs

```diff
@@ -234,6 +235,70 @@ impl ResolveFilter for ExternalFilter {
     }
 }
 
+impl ResolveFilter for FirstFilter {
+    fn resolve<'t, 'py>(
+        &self,
+        variable: Option<Content<'t, 'py>>,
+        py: Python<'py>,
+        _template: TemplateString<'t>,
+        _context: &mut Context,
+    ) -> ResolveResult<'t, 'py> {
+        let content = match variable {
+            Some(content) => {
+                // Try to get the first item from the content
+                match content {
+                    Content::Py(obj) => {
+                        // Try to get the first item using Python's indexing
+                        // Django only catches IndexError, not TypeError
+                        match obj.get_item(0) {
+                            Ok(first) => Some(Content::Py(first)),
+                            Err(e) => {
+                                // Check if it's an IndexError
+                                if e.is_instance_of::<pyo3::exceptions::PyIndexError>(py) {
+                                    // IndexError -> return empty string
+                                    "".as_content()
+                                } else {
+                                    // Other errors (like TypeError) -> propagate
+                                    return Err(e.into());
+                                }
+                            }
```

Comment:
I'd import `PyIndexError` at the top of the module. We can also reduce the verbosity by inlining the `isinstance` check into the `match`:

```suggestion
                            Err(e) if e.is_instance_of::<PyIndexError>(py) => "".as_content(),
                            Err(e) => return Err(e.into()),
```